### ISSUE TITLE
Component on same like as text, results in compile error

### ISSUE THREAD

**Issue by Perusae** [a-h/templ#571]
The following code (what used to work with the old {! icon } syntax, break with the new @ syntax

Old version

```
if icon != nil {
	<svg> {! icon } </svg>
}
```
Result in: works without issue

```
if icon != nil {
	<svg> @icon() </svg>
}
```
Results in: error=failed to generate code parsing error: if: expected nodes, but none were found

```
if icon != nil {
	<svg>
              @icon()
      </svg>
}
```
Result in: works without issue

**Comment by a-h** [comment]
I assume this was on the main branch?

I spotted the problem yesterday, but hadn't committed my changes. I didn't quite fix it in the 24 minutes since you posted the issue! 😁 

Please re-open if it's a wider issue.

**Comment by a-h** [comment]
I'm shaking down some more bugs out of the migration to the Go AST expression parser which is taking some time.

I wrote some fuzzing tests, which helped, but I also downloaded 1000 templ projects from Github, and am running `templ generate` on them all and working through issues that I find.

The end result is multiline Go statements in templ files, and should make defining variables in templ files possible too.

But... it's a big risky change, as you've noticed!
